### PR TITLE
Run Github actions to validate new PRs

### PR DESCRIPTION
**Pull request by meijiao**
[validate-PR](https://github.com/jup-ag/token-list/actions/workflows/validate-PR.yml)

Gets git diff of change in PR https://github.com/jup-ag/token-list/blob/352b9b63840a5e3107ac9d74c8e7fadef6eed4d9/src/main.ts#L14 

Checks for: https://github.com/jup-ag/token-list/blob/352b9b63840a5e3107ac9d74c8e7fadef6eed4d9/src/validate.ts#L11
- Changes to unrelated files
- Removals
- Duplicates against existing strict list https://github.com/jup-ag/token-list/blob/352b9b63840a5e3107ac9d74c8e7fadef6eed4d9/src/get-validated.ts#L13
- Invalid inputs

Todo:
- Check against onchain metadata
- Turn on checks for unrelated file change after this PR goes in

### CODE REVIEW

File: src/get-validated.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { parse } from "csv-parse/sync";
```
Comment (ZhengYuTay) [inline src/get-validated.ts:1]: dont think this is used

File: src/validate.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  if (validatedSet.symbols.has(symbol)) {
+      errors.push(ValidationError.DUPLICATE_SYMBOL);
+  }
+  
+  if (validatedSet.mints.has(mint)) {
+      errors.push(ValidationError.DUPLICATE_MINT);
+  }
+
+  // Flag invalid mint address
+  const TOKEN_MINT = new PublicKey(mint);
```
Comment (ZhengYuTay) [inline src/validate.ts:54]: this would throw error if their token mint is invalid, maybe a try catch might be needed

File: src/validate.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  if(!PublicKey.isOnCurve(new PublicKey(TOKEN_MINT))) {
```
Comment (ZhengYuTay) [inline src/validate.ts:55]: not sure we need this, but nice touch, we can even do a getAccountInfo to check if the account is on chain, but that's optional.

TOKEN_MINT is PublicKey already, no need to warp again

File: src/get-validated.ts (hunk 1 above)
Comment (meijiao) [reply src/get-validated.ts]: removed

File: src/validate.ts (hunk 2 above)
Comment (meijiao) [reply src/validate.ts]: fixed and tested with invalid mint, will not throw.

File: src/validate.ts (hunk 3 above)
Comment (meijiao) [reply src/validate.ts]: fixed

**Pull request merged**